### PR TITLE
Render Markdown emphasis in clue text

### LINKED ISSUE #460: markdown in clue text
Image

Image

### PR DESCRIPTION
**Pull request by ScaleOvenStove**
## Summary
- Clues like the NYT puzzle's `Put all the b*o*l*d* letters in this clue t*o*gethe*r*?` were rendering asterisks literally, breaking the meta clue (fixes #460)
- `ClueText` now expands `**text**` → `<strong>` and `*text*` → `<em>` before HTML parsing
- CommonMark-style guard rails so stray asterisks (e.g. `5 * 6`) aren't treated as emphasis

## Test plan
- [x] New `ClueText.test.tsx` covers plain text, HTML, double-quote italics, single + double Markdown spans, the multi-span clue from the issue, and asterisk edge cases
- [x] `pnpm test` — 382 tests pass
- [x] `pnpm eslint --max-warnings 0 src/`
- [x] `pnpm prettier --check`
- [x] `pnpm tsc --noEmit`

https://claude.ai/code/session_011sJYM4UyFUo8YKwZwMjWzu

### CODE REVIEW

File: src/components/Player/ClueText.ts [hunk 1]
```diff
@@ -3,6 +3,17 @@ import type {JSX} from 'react';
...
+// Convert Markdown emphasis markers to HTML so the existing HTML parser
+// can render them. `**text**` becomes <strong>, `*text*` becomes <em>.
+// Bold is processed first so that `**…**` wins over `*…*`. The character
+// adjacent to each marker must be non-whitespace, matching CommonMark
+// emphasis rules — this prevents stray asterisks (e.g. "5 * 6 = 30") from
+// being treated as emphasis.
+const applyMarkdown = (text: string): string =>
+  text
+    .replace(/\*\*([^*\s](?:[^*]*?[^*\s])?)\*\*/g, '<strong>$1</strong>')
+    .replace(/\*([^*\s](?:[^*]*?[^*\s])?)\*/g, '<em>$1</em>');
```
[the remark this answers is not in the document]
Comment (ScaleOvenStove) [reply src/components/Player/ClueText.ts]: Thanks for the review. I tried this change and it actually breaks the primary use case from #460: `b*o*l*d*` renders as `b<em>o*l</em>d*` instead of `b<em>o</em>l<em>d</em>`, because `.*?` (even non-greedy) is happy to swallow the interior asterisks once the outer optional group is preferred. I verified this via the existing "renders multiple Markdown spans within a single clue" test.

The `[^*]*?` pattern is intentional — it's what forces per-pair matching for `*x*y*z*` sequences (single-letter emphasis, which is the shape NYT-style meta clues use). The tradeoff is that a literal stray `*` can't appear inside emphasis, but that's rare in crossword clues and the opposite tradeoff would regress the reported bug.

---
_Generated by [Claude Code](https://claude.ai/code)_
